src/components/controls/PasswordInput/PasswordInput.tsx

```diff
@@ -0,0 +1,119 @@
+'use client';
+
+import React from 'react';
+
+import {Eye, EyeSlash} from '@gravity-ui/icons';
+
+import {useControlledState} from '../../../hooks';
+import {ActionTooltip} from '../../ActionTooltip';
+import {Button} from '../../Button';
+import {ClipboardButton} from '../../ClipboardButton';
+import {Icon} from '../../Icon';
+import {block} from '../../utils/cn';
+import {TextInput} from '../TextInput';
+import type {TextInputProps} from '../TextInput';
+
+import {i18n} from './i18n';
+import {getActionButtonSizeAndIconSize} from './utils';
+
+import './PasswordInput.scss';
+
+const b = block('password-input');
+
+export type PasswordInputProps = Omit<TextInputProps, 'type'> & {
+    /** Show copy button */
+    showCopyButton?: boolean;
+    /** Show reveal button */
+    showRevealButton?: boolean;
+    /** Disable the tooltip for the copy button. The tooltip will not be displayed */
+    hasCopyTooltip?: boolean;
+    /** Disable the tooltip for the reveal button. The tooltip will not be displayed */
+    hasRevealTooltip?: boolean;
+    /** Determines the visibility state of the password input field */
+    revealValue?: boolean;
+    /** A callback function that is invoked whenever the revealValue state changes */
+    onRevealValueUpdate?: (value: boolean) => void;
+};
+
+export const PasswordInput = (props: PasswordInputProps) => {
+    const {
+        autoComplete,
+        showCopyButton,
+        rightContent,
+        endContent,
+        showRevealButton,
+        size = 'm',
+        hasCopyTooltip = true,
+        hasRevealTooltip = true,
```

Comment:
@amje @korvin89 What do we want to do here, remove default values, rename props or do nothing?

Reply:
I suggest
- to rename ~~hasCopyTooltip~~ to `showCopyTooltip` and remove default. I think there is no need in this element by default
- to rename ~~hasRevealTooltip~~ to `showRevealTooltip` and remove default. I think we can hide this eleement by default

Reply:
fixed